MiniAn/minian_main.py

```diff
@@ -183,19 +194,133 @@ def preview(minian_parameters):
     cluster.close()
 
 
+def penalties_preview(minian_params):
+    """
+    Saves penalties_preview
+    """
+
+    # We assume that initialization preview was generated first
+    # Read the saved data during the generation of the Initialization Preview
+    # It's in the "[TmpDir]/Intermediate", e.g. Y_hw_chk, Y_fm_chk
+
+    # Start cluster
+    print(mn_defs.Messages.START_CLUSTER, flush=True)
+    cluster = LocalCluster(**minian_params.params_LocalCluster)
+    annt_plugin = TaskAnnotation()
+    cluster.scheduler.add_plugin(annt_plugin)
+    client = Client(cluster)
+
+    intpath = os.path.join(minian_params.paths[defs.Parameters.Path.TMP_DIR], mn_defs.Folder.INTERMEDIATE)
+    subset = {"frame": slice(*minian_params.preview_params[defs.Parameters.Preview.RANGE])}
+
+    minian_ds  = open_minian(intpath)
+    Y_hw_chk = minian_ds["Y_hw_chk"]
+    Y_fm_chk = minian_ds["Y_fm_chk"]
+
+    file_, chk, _ = load_chunk(intpath, subset, minian_params)
+    file_.close()
+
+    with h5py.File(minian_params.preview_params[defs.Parameters.Preview.FILEPATH], 'r') as hdf5_file:
+        initialization_group = hdf5_file[mn_defs.Preview.Group.INITIALIZATION]
+        seeds_dataset = initialization_group[mn_defs.Preview.Dataset.SEEDS]
+        mask_mrg      = list(seeds_dataset.attrs[mn_defs.Preview.Attribute.MERGED])
+        seeds         = pd.DataFrame({
+            "width": np.array(seeds_dataset)[0,:],
+            "height": np.array(seeds_dataset)[1,:],
+            "mask_mrg":[k in mask_mrg for k in range(seeds_dataset.attrs[mn_defs.Preview.Attribute.SEED_COUNT])]
```

Comment:
mask_mrg should `True` or `False` -- true if the index is in the `seeds_dataset.attrs[mn_defs.Preview.Attribute.MERGED]`, and false if not. Please load the attributes on a separate line.